pkg/webhook/controlplane/ensurer.go

```diff
@@ -88,8 +88,9 @@ func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, gctx gconte
 		return err
 	}
 
+	k8sGreaterEqual127, _ := versionutils.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, ">=", "1.27")
 	if c := extensionswebhook.ContainerWithName(ps.Containers, "kube-apiserver"); c != nil {
-		ensureKubeAPIServerCommandLineArgs(c, csiMigrationCompleteFeatureGate)
+		ensureKubeAPIServerCommandLineArgs(c, k8sGreaterEqual127, csiMigrationCompleteFeatureGate)
 		ensureEnvVars(c)
```

Comment:
I would rather prefer creation of semver version and usage of predefined constraints instead of usage of `versionutils.CompareVersions`.
See https://github.com/gardener/gardener-extension-provider-openstack/pull/551/files#diff-cc0552e76e65ffff14fa4d53e143f0d325fb62ff3c926693a9153658dcbfd23dR190.

This comment applies for all the similar changes in this file.